code/datums/quirks/positive_quirks/spacer.dm

```diff
@@ -0,0 +1,195 @@
+		/obj/item/reagent_containers/pill/gravitum,
+	)
+
+	/// How long on a planet before we get averse effects
+	var/planet_period = 3 MINUTES
+	/// TimerID for time spend on a planet
+	var/planetside_timer
+	/// How long in space before we get beneficial effects
+	var/recover_period = 1 MINUTES
+	/// TimerID for time spend in space
+	var/recovering_timer
+
+/datum/quirk/spacer_born/add(client/client_source)
+	// Only start tracking when we arrive on station (or a mining level of a map is weird)
+	if(!start_tracking(quirk_holder))
+		RegisterSignal(quirk_holder, COMSIG_MOVABLE_Z_CHANGED, PROC_REF(arrived_at_station))
+
+/datum/quirk/spacer_born/add_unique(client/client_source)
+	// drift slightly faster through zero G
+	quirk_holder.inertia_move_delay *= 0.8
+
+	var/mob/living/carbon/human/human_quirker = quirk_holder
+	human_quirker.set_mob_height(HUMAN_HEIGHT_TALLER)
+	human_quirker.physiology.pressure_mod *= 0.8
+	human_quirker.physiology.cold_mod *= 0.8
+
+/datum/quirk/spacer_born/post_add()
+	var/on_a_planet = SSmapping.config.planetary
+	var/planet_job = istype(quirk_holder.mind?.assigned_role, /datum/job/shaft_miner)
+	if(!on_a_planet && !planet_job)
+		return
+	var/datum/bank_account/spacer_account = quirk_holder.get_bank_account()
+	if(!isnull(spacer_account))
+		spacer_account.payday_modifier *= 1.25
+		to_chat(quirk_holder, span_info("Given your background as a Spacer, \
+			you are awarded with a 25% hazard pay bonus due to your [on_a_planet ?  "station" : "occupational"] assignment."))
+
+	// Supply them with some patches to help out on their new assignment
+	var/obj/item/storage/pill_bottle/ondansetron/disgust_killers = new()
+	disgust_killers.desc += " Best to take one when travelling to a planet's surface."
+	quirk_holder.equip_to_slot_if_possible(disgust_killers, ITEM_SLOT_BACKPACK, qdel_on_fail = TRUE, initial = TRUE, indirect_action = TRUE)
+
+/datum/quirk/spacer_born/remove()
+	UnregisterSignal(quirk_holder, COMSIG_MOVABLE_Z_CHANGED)
+
+	if(QDELING(quirk_holder))
+		return
+
+	quirk_holder.inertia_move_delay /= 0.8
+	quirk_holder.clear_mood_event("spacer")
+	quirk_holder.remove_movespeed_modifier(/datum/movespeed_modifier/spacer)
+	quirk_holder.remove_status_effect(/datum/status_effect/spacer)
+
+	var/mob/living/carbon/human/human_quirker = quirk_holder
+	human_quirker.set_mob_height(HUMAN_HEIGHT_MEDIUM)
+	human_quirker.physiology.pressure_mod /= 0.8
+	human_quirker.physiology.cold_mod /= 0.8
+
+/datum/quirk/spacer_born/proc/start_tracking(mob/living/tracking)
+	var/new_z = tracking.loc?.z
+	// Some maps may decide to spawn people on mining Z, so just cast a wide net.
+	if(!is_station_level(new_z) && !is_mining_level(new_z) && !is_centcom_level(new_z))
+		return FALSE
+
+	UnregisterSignal(quirk_holder, COMSIG_MOVABLE_Z_CHANGED) // gets rid of arrived_at_station registration if present
+	RegisterSignal(quirk_holder, COMSIG_MOVABLE_Z_CHANGED, PROC_REF(spacer_moved)) // replaces it with spacer_moved
```

Comment:
why do we not just register onto spacer_moved at the start? aren't the checks in this proc identical to what's inside is_on_a_planet (and thus spacer_moved)